# Which week in the last year had the most number of commits?

In [15]:
import requests
import json
import pandas as pd

url= 'https://api.github.com/repos/d3/d3/stats/commit_activity'
r = requests.get(url)


In [14]:
import time
cols = ['week','total_commits']
df = pd.DataFrame(columns=cols)
for i, x in enumerate(r.json()):
    df.loc[i, :] = x['week'],x['total']
    
df['week']= df['week'].map(lambda x: time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(x)))
print df[df['total_commits']==df['total_commits'].max()]

                   week  total_commits
48  2018-01-27 16:00:00            6.0


The week of '2018-01-27 16:00:00' had the most commits with 6.

# How many commits did the other contributors make?

In [22]:
url= 'https://api.github.com/repos/d3/d3/stats/contributors'
r = requests.get(url)

cols = ['author_id', 'total_commits']
df = pd.DataFrame(columns=cols)
for i, x in enumerate(r.json()):
    df.loc[i,:] = x['author']['login'], x['total']

print "There are "+ str(len(df)) +" contributors including owner 'mbostock' "



There are 100 contributors including owner 'mbostock' 


In [23]:
mbostock_commits = df[df['author_id']=='mbostock']['total_commits'].values[0]
print "Owner mbostock made "+str(mbostock_commits)+" commmits."

jasondavies_commits = df[df['author_id']=='jasondavies']['total_commits'].values[0]

print df[df['total_commits']>10]

Owner mbostock made 2319 commmits.
       author_id total_commits
93        natevw            11
94  larskotthoff            11
95        yasirs            11
96      27359794            24
97    kitmonisit            50
98   jasondavies           755
99      mbostock          2319



Aside from the owner, there are only 6 other contributors with over 10 total commits. Contributor 'jasondavies' notably had 755, by far the most of the contributors. 


In [24]:
filter_authors = [a for a in df['author_id'] if a not in ['jasondavies','mbostock']]
other_contrib_commmitstotal = df[df['author_id'].isin(filter_authors)]['total_commits'].sum()
print other_contrib_commmitstotal

323
